### PR TITLE
Feature/prime 692 decline

### PR DESCRIPTION
**Pull request by hogmann**
Currently just need to hook up the guard, but may be taking the wrong route at which point i should route to the "Declined" Screen so ill ask some advise in the morning.

### CODE REVIEW

File: prime-dotnet-webapi/Services/EnrolleeService.cs [hunk 1]
```diff
@@ -434,8 +434,19 @@ public async Task<IEnumerable<HpdidLookup>> HpdidLookupAsync(IEnumerable<string>
             return await _context.Enrollees
                 .Include(e => e.AccessTerms)
                 .Where(e => hpdids.Contains(e.HPDID))
+                .Where(e => e.CurrentStatus.StatusCode != (int)StatusType.Declined)
```
Comment (james-hollinger) [inline prime-dotnet-webapi/Services/EnrolleeService.cs:437]: Status has an IsType method

File: prime-angular-frontend/src/app/app.routes.ts [hunk 2]
```diff
@@ -5,4 +5,5 @@ export class AppRoutes {
   public static PAGE_NOT_FOUND = 'page-not-found';
   public static HELP = 'help';
   public static DEFAULT = '**';
+  public static ACCESS_DECLINED = 'access-declined';
```
Comment (mtpultz) [inline prime-angular-frontend/src/app/app.routes.ts:8]: Feels like this is left over from your refactor, and now exists in the EnrolmentRoutes

File: prime-angular-frontend/src/app/modules/adjudication/shared/components/adjudication-container/adjudication-container.component.ts [hunk 3]
```diff
@@ -231,6 +231,65 @@ export class AdjudicationContainerComponent extends AbstractComponent implements
       });
   }
 
+  public onDeclineEnrollee(enrolleeId: number) {
+    const data: DialogOptions = {
+      title: 'Decline Enrollee',
+      message: 'When declined the enrollee will not have access to PRIME and their Terms of Access revoked,'
+        + ' Are you sure you want to lock this enrollee ? ',
```
Comment (mtpultz) [inline prime-angular-frontend/src/app/modules/adjudication/shared/components/adjudication-container/adjudication-container.component.ts:238]: You can get multiple lines using backticks and don't have to concatenate.  Also, might be worth updating "and their Terms of Access revoked" to "and their Terms of Access will be revoked"

File: prime-angular-frontend/src/app/modules/enrolment/pages/access-declined/access-declined.component.html [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+  <app-page-header>PharmaNet Enrolment</app-page-header>
+
+  <app-enrolment-progress-indicator *ngIf="isInitialEnrolment"
+                                    message="Enrolment Declined"
+                                    [currentRoute]="EnrolmentRoutes.ACCESS_DECLINED"></app-enrolment-progress-indicator>
+
+  <div class="mb-5">
+    <app-alert type="danger">
+      <ng-container #alertContent
```
Comment (mtpultz) [inline prime-angular-frontend/src/app/modules/enrolment/pages/access-declined/access-declined.component.html:11]: Can you update the alert to have a title like the access-locked page "Your PRIME enrolment has been declined", and it the content of the alert looks like it should be "Your application for PharmaNet access has not be approved."

File: prime-dotnet-webapi/Services/EnrolleeService.cs [hunk 5]
```diff
@@ -434,8 +434,19 @@ public async Task<IEnumerable<HpdidLookup>> HpdidLookupAsync(IEnumerable<string>
...
                 .Select(e => HpdidLookup.FromEnrollee(e))
                 .ToListAsync();
         }
+
+        public async Task<Enrollee> GetEnrolleeForUserIdNotDeclinedAsync(Guid userId)
```
Comment (james-hollinger) [inline prime-dotnet-webapi/Services/EnrolleeService.cs:442]: I think I would rather see an optional parameter on getenrolleeforuserid with excludeDeclined = false or something like that.

**Pull request merged**